std/stdio.d

```diff
@@ -2877,13 +2877,27 @@ $(D Range) that locks the file and allows fast writing to it.
                     if (result != writeme.length) errnoEnforce(0);
                     return;
                 }
-            }
 
-            // put each element in turn.
-            alias Elem = Unqual!(ElementType!A);
-            foreach (Elem c; writeme)
+                // put each element in turn.
+                for (; !writeme.empty; writeme.popFront)
```

Comment:
No. As pointed out in the discussion, LDC/GDC are smart enough.
The code should be improved to be more high-level and abstract, not the other way around.

Have a look at what LDC is doing: https://godbolt.org/g/HuhD6T
It even inlines `put` completely:

```asm
void example.put(immutable(char)[]):
  mov rcx, qword ptr [rip + int example.put(immutable(char)[]).__foreachbody2(ref dchar)@GOTPCREL]
  xor edx, edx
  jmp _aApplycd1@PLT

int example.put(immutable(char)[]).__foreachbody2(ref dchar):
  push r14
  push rbx
  push rax
  mov edi, dword ptr [rsi]
  cmp edi, 127
  ja .LBB1_2
  xor esi, esi
  call fputc_unlocked@PLT
  jmp .LBB1_6
.LBB1_2:
  lea rsi, [rsp + 4]
  call _D3std3utf__T6encodeVEQu8typecons__T4FlagVAyaa19_7573655265706c6163656d656e744463686172ZQCai0ZQDdFNaNfJG4awZm@PLT
  mov r14, rax
  test r14, r14
  je .LBB1_6
  xor ebx, ebx
.LBB1_4:
  cmp rbx, 3
  ja .LBB1_7
  movzx edi, byte ptr [rsp + rbx + 4]
  xor esi, esi
  call fputc_unlocked@PLT
  inc rbx
  cmp rbx, r14
  jb .LBB1_4
.LBB1_6:
  xor eax, eax
  add rsp, 8
  pop rbx
  pop r14
  ret
.LBB1_7:
  lea rsi, [rip + .L.str]
  mov edi, 63
  mov edx, 39
  call _d_arraybounds@PLT
```